test/functional/test_runner.py

```diff
@@ -236,26 +239,24 @@ def main():
     if tests:
         # Individual tests have been specified. Run specified tests that exist
         # in the ALL_SCRIPTS list. Accept the name with or without .py extension.
-        tests = [re.sub("\.py$", "", t) + ".py" for t in tests]
+        tests = [re.sub("\.py$", "", test) + ".py" for test in tests]
         test_list = []
-        for t in tests:
-            if t in ALL_SCRIPTS:
-                test_list.append(t)
+        for test in tests:
+            if test in ALL_SCRIPTS:
+                test_list.append(test)
             else:
-                print("{}WARNING!{} Test '{}' not found in full test list.".format(BOLD[1], BOLD[0], t))
+                print("{}WARNING!{} Test '{}' not found in full test list.".format(BOLD[1], BOLD[0], test))
+    elif args.extended:
+        # Include extended tests
+        test_list = ALL_SCRIPTS
     else:
-        # No individual tests have been specified.
-        # Run all base tests, and optionally run extended tests.
+        # Run base tests only
         test_list = BASE_SCRIPTS
```

Comment:
Could you change this to `test_list += BASE_SCRIPTS`, otherwise later modification of `test_list` will also modify the const BASE_SCRIPTS.

Reply:
I missed that, thank you. [PR opened](https://github.com/bitcoin/bitcoin/pull/12710)